Assets/Packages/com.studio23.ss2.bettercursormanager/Runtime/Core/BetterCursor.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using Studio23.SS2.BetterCursor.Data;
+using UnityEngine;
+
+namespace Studio23.SS2.BetterCursor.Core
+{
+    public class BetterCursor : MonoBehaviour
+    {
+        public static BetterCursor Instance;
+
+        [SerializeField] private Canvas _canvas;
+        [SerializeField] private Camera _camera;
+        public CursorData CurrentCursor;
+
+        public bool UiOnHoverEnabled;
+        [SerializeField] private LayerMask _onHoverMask;
+
+        private CursorAnimationController _animationController;
+        private CursorLocoMotionController _locoMotionController;
+        private CursorEventController _eventController;
+
+        private void Awake()
+        {
+            if (Instance == null)
+                Instance = this;
+            else
+                Destroy(gameObject);
+        }
+
+        private void Start()
+        {
+            Cursor.visible = false;
+
+            _camera = _camera ?? Camera.main;
+            if (_camera == null)
+                Debug.LogError(
+                    "No Camera assigned on Better Cursor.It must be assigned for it to work.Or Tag a camera as MainCamera");
+
+            _canvas = GetComponent<Canvas>();
+            _eventController = GetComponentInChildren<CursorEventController>();
+            _locoMotionController = GetComponentInChildren<CursorLocoMotionController>();
+            _animationController = GetComponentInChildren<CursorAnimationController>();
+            Initialize();
+            ChangeCursor(CurrentCursor);
+            ChangeCursorLockState(false);
+        }
+
+
+        private void Initialize()
+        {
+            if (CurrentCursor == null) CurrentCursor = Resources.Load<CursorData>("Default Cursor");
+
+            _locoMotionController.Initialize(_canvas, CurrentCursor);
+            _animationController.Initialize(CurrentCursor);
+            _eventController.Initialize(_onHoverMask, _camera);
```

Comment:
_onHoverMask field is not set.  from cursordata. I recommend removing the field and just sending cursordata.hovermask